node/src/types/chainspec/core_config.rs

```diff
@@ -84,6 +84,9 @@ pub struct CoreConfig {
 
     /// Which consensus protocol to use.
     pub consensus_protocol: ConsensusProtocolName,
+
+    /// The maximum amount of delegators per validator.
+    pub max_delegators_per_validator: Option<u32>,
```

Comment:
I don't like having optional fields in the chainspec, since it's too easy to forget to update the chainspec.toml (as has happened in this PR).

I'd suggest making this non-optional here, and treating a value of `0` as `None` when converting to the corresponding engine_config field.

Either way - if we decide against that suggestion, I strongly think we should add entries to `resources/production/chainspec.toml` and `resources/local/chainspec.toml.in` along with a note about the default value if not provided (in the case we stick leaving this optional) or a note about setting it to `0` meaning it's disabled.

Reply:
I like the idea of having the zero value corresponding to `None`, I will make that change

Reply:
Added the change as requested